src/Provider/Provider.Web/Orchestrators/Reports/ProviderApplicationsReportOrchestrator.cs

```diff
@@ -49,23 +50,23 @@ public Task<Guid> PostCreateViewModelAsync(ProviderApplicationsReportCreateEditM
             switch (model.DateRange)
             {
                 case DateRangeType.Last7Days:
-                    fromDate = toDateInclusive.Date.AddDays(-7);
+                    fromDate = _timeProvider.Today.AddDays(-7);
                     break;
                 case DateRangeType.Last14Days:
-                    fromDate = toDateInclusive.Date.AddDays(-14);
+                    fromDate = _timeProvider.Today.AddDays(-14);
                     break;
                 case DateRangeType.Last30Days:
-                    fromDate = toDateInclusive.Date.AddDays(-30);
+                    fromDate = _timeProvider.Today.AddDays(-30);
                     break;
                 case DateRangeType.Custom:
-                    fromDate = model.FromDate.AsDateTimeUk().Value.ToUniversalTime();
-                    toDateInclusive = model.ToDate.AsDateTimeUk().Value.ToUniversalTime().AddDays(1).AddTicks(-1);
+                    fromDate = model.FromDate.AsDateTimeUk(DateTimeStyles.AssumeLocal).Value.ToUniversalTime();
+                    toDateInclusive = model.ToDate.AsDateTimeUk(DateTimeStyles.AssumeLocal).Value.ToUniversalTime().AddDays(1).AddTicks(-1);
```

Comment:
I'm just a bit confused here! What is happening here? Will having some comments help?